cpp/include/cudf/io/parquet_schema.hpp

```diff
@@ -27,55 +25,221 @@
+  INT_16           = 16,  // A signed integer 16-bit value as INT32
+  INT_32           = 17,  // A signed integer 32-bit value as INT32
+  INT_64           = 18,  // A signed integer 8-bit value as INT64
+  JSON             = 19,  // A JSON document embedded within a single UTF8 column.
+  BSON             = 20,  // A BSON document embedded within a single BINARY column.
+  INTERVAL = 21,  // This type annotates a time interval stored as a FIXED_LEN_BYTE_ARRAY of length
+                  // 12 for 3 integers {months,days,milliseconds}
+  NA = 25,        // No Type information, For eg, all-nulls.
+};
+
+/**
+ * @brief Encoding types for the actual data stream
+ */
+enum class Encoding : uint8_t {
+  PLAIN                   = 0,
+  GROUP_VAR_INT           = 1,  // Deprecated, never used
+  PLAIN_DICTIONARY        = 2,
+  RLE                     = 3,
+  BIT_PACKED              = 4,  // Deprecated by parquet-format in 2013, superseded by RLE
+  DELTA_BINARY_PACKED     = 5,
+  DELTA_LENGTH_BYTE_ARRAY = 6,
+  DELTA_BYTE_ARRAY        = 7,
+  RLE_DICTIONARY          = 8,
+  BYTE_STREAM_SPLIT       = 9,
+  NUM_ENCODINGS           = 10,
+};
+
+/**
+ * @brief Compression codec used for compressed data pages
+ */
+enum Compression {
+  UNCOMPRESSED = 0,
+  SNAPPY       = 1,
+  GZIP         = 2,
+  LZO          = 3,
+  BROTLI       = 4,  // Added in 2.3.2
+  LZ4          = 5,  // deprecated; based on LZ4, but with an additional undocumented framing scheme
+  ZSTD         = 6,  // Added in 2.3.2
+  LZ4_RAW      = 7,  // "standard" LZ4 block format
+};
 
-constexpr uint32_t parquet_magic = (('P' << 0) | ('A' << 8) | ('R' << 16) | ('1' << 24));
+/**
+ * @brief Compression codec used for compressed data pages
+ */
+enum FieldRepetitionType {
+  NO_REPETITION_TYPE = -1,
+  REQUIRED = 0,  // This field is required (can not be null) and each record has exactly 1 value.
+  OPTIONAL = 1,  // The field is optional (can be null) and each record has 0 or 1 values.
+  REPEATED = 2,  // The field is repeated and can contain 0 or more values
+};
+
+/**
+ * @brief Types of pages
+ */
+enum class PageType : uint8_t {
+  DATA_PAGE       = 0,
+  INDEX_PAGE      = 1,
+  DICTIONARY_PAGE = 2,
+  DATA_PAGE_V2    = 3,
+};
+
+/**
+ * @brief Enum to annotate whether lists of min/max elements inside ColumnIndex
+ * are ordered and if so, in which direction.
+ */
+enum BoundaryOrder {
+  UNORDERED  = 0,
+  ASCENDING  = 1,
+  DESCENDING = 2,
+};
+
+/**
+ * @brief Thrift compact protocol struct field types
+ */
+enum class FieldType : uint8_t {
+  BOOLEAN_TRUE  = 1,
+  BOOLEAN_FALSE = 2,
+  I8            = 3,
+  I16           = 4,
+  I32           = 5,
+  I64           = 6,
+  DOUBLE        = 7,
+  BINARY        = 8,
+  LIST          = 9,
+  SET           = 10,
+  MAP           = 11,
+  STRUCT        = 12,
+  UUID          = 13,
+};
 
 /**
  * @brief Struct that describes the Parquet file data header
  */
 struct file_header_s {
+  // Parquet 4-byte magic number "PAR1"
```

Comment:
Add docs for a bunch of variables/functions in this file to pass the `style-check` in CI